log-collector/src/main/scala/io/phdata/pulse/logcollector/Util.scala

```diff
@@ -0,0 +1,34 @@
+package io.phdata.pulse.logcollector
+
+import io.phdata.pulse.common.domain.LogEvent
+
+object Util {
+
+  /**
+   * Convert a [[LogEvent]] to a flattened [[Map[String,String]]
+   * @param logEvent LogEvent to convert
+   * @return The event, with properties flattened, as a [[Map[String,String]]
+   */
+  def logEventToFlattenedMap(logEvent: LogEvent): Map[String, String] = {
+    val values = logEvent.productIterator
+    val logEventmap = logEvent.getClass.getDeclaredFields.map {
+      _.getName -> (values.next() match {
+        case x => x
+      })
+    }.toMap
+    logEventmap
+      .flatten {
+        case ((key, map: Option[Map[String, String]]))
+            if map.getOrElse(None).isInstanceOf[Map[String, String]] =>
```

Comment:
I don't think this code should necessarily stay for the reasons above, but all the casting shouldn't be necessary.
Since option has a map type, it might simplify to something more like

```
case ((_, properties: Some[Map[String, String]])) => properties.get
```

`isInstanceOf` is almost always a smell, especially since you can pattern match on types.